extensions/standard-processors/tests/unit/PutTCPTests.cpp

```diff
@@ -0,0 +1,454 @@
+  ssl_data.key_loc = (executable_dir / "resources/localhost_by_A.pem").string();
+  return ssl_data;
+}
+}  // namespace
+
+class PutTCPTestFixture {
+ public:
+  PutTCPTestFixture() {
+    LogTestController::getInstance().setTrace<PutTCP>();
+    LogTestController::getInstance().setInfo<core::ProcessSession>();
+    LogTestController::getInstance().setTrace<utils::net::Server>();
+    put_tcp_->setProperty(PutTCP::Hostname, "${literal('localhost')}");
+    put_tcp_->setProperty(PutTCP::Port, utils::StringUtils::join_pack("${literal('", std::to_string(port_), "')}"));
+    put_tcp_->setProperty(PutTCP::Timeout, "200 ms");
+    put_tcp_->setProperty(PutTCP::OutgoingMessageDelimiter, "\n");
+  }
+
+  ~PutTCPTestFixture() {
+    stopServer();
+  }
+
+  void startTCPServer() {
+    gsl_Expects(!listener_ && !server_thread_.joinable());
+    listener_ = std::make_unique<SessionAwareTcpServer>(std::nullopt, port_, core::logging::LoggerFactory<utils::net::Server>().getLogger());
+    server_thread_ = std::thread([this]() { listener_->run(); });
+  }
+
+  void startSSLServer() {
+    gsl_Expects(!listener_ && !server_thread_.joinable());
+    listener_ = std::make_unique<SessionAwareSslServer>(std::nullopt,
+        port_,
+        core::logging::LoggerFactory<utils::net::Server>().getLogger(),
+        createSslDataForServer(),
+        utils::net::SslServer::ClientAuthOption::REQUIRED);
+    server_thread_ = std::thread([this]() { listener_->run(); });
+  }
+
+  void stopServer() {
+    if (listener_)
+      listener_->stop();
+    if (server_thread_.joinable())
+      server_thread_.join();
+    listener_.reset();
+  }
+
+  size_t getNumberOfActiveSessions() {
+    if (auto session_aware_listener = dynamic_cast<ISessionAwareServer*>(listener_.get())) {
+      return session_aware_listener->getNumberOfSessions() - 1;  // There is always one inactive session waiting for a new connection
+    }
+    return -1;
+  }
+
+  void closeActiveConnections() {
+    if (auto session_aware_listener = dynamic_cast<ISessionAwareServer*>(listener_.get())) {
+      session_aware_listener->closeSessions();
+    }
+    std::this_thread::sleep_for(200ms);
+  }
+
+  auto trigger(const std::string_view& message) {
+    return controller_.trigger(message);
+  }
+
+  auto getContent(const auto& flow_file) {
+    return controller_.plan->getContent(flow_file);
+  }
+
+  std::optional<utils::net::Message> tryDequeueReceivedMessage() {
+    auto timeout = 200ms;
+    auto interval = 10ms;
+
+    auto start_time = std::chrono::system_clock::now();
+    utils::net::Message result;
+    while (start_time + timeout > std::chrono::system_clock::now()) {
+      if (listener_->tryDequeue(result))
+        return result;
+      std::this_thread::sleep_for(interval);
+    }
+    return std::nullopt;
+  }
+
+  void addSSLContextToPutTCP(const std::filesystem::path& ca_cert, std::optional<std::filesystem::path> client_cert_key) {
```

Comment:
the second parameter should also be `const &`

Reply:
good idea changed it in https://github.com/apache/nifi-minifi-cpp/pull/1419/commits/9928eaa75f607d1d337d289c7bbf84f23e20ee5a#diff-bbb56b112b398a6450c23e71e5dbbfb17bf20dad3d2fb908a1c6de26118f9ee8R191